bftengine/src/bftengine/messages/ClientRequestMsg.cpp

```diff
@@ -64,29 +86,107 @@ ClientRequestMsg::ClientRequestMsg(ClientRequestMsgHeader* body)
 bool ClientRequestMsg::isReadOnly() const { return (msgBody()->flags & READ_ONLY_REQ) != 0; }
 
 void ClientRequestMsg::validate(const ReplicasInfo& repInfo) const {
-  ConcordAssert(senderId() != repInfo.myId());
-  if (size() < sizeof(ClientRequestMsgHeader) ||
-      size() < (sizeof(ClientRequestMsgHeader) + msgBody()->requestLength + msgBody()->cidLength + spanContextSize()))
-    throw std::runtime_error(__PRETTY_FUNCTION__);
+  uint16_t expectedSigLen = getExpectedSignatureLength();
+  validateRequest(repInfo, expectedSigLen);
+  if (expectedSigLen > 0) {
+    validateRequestSignature(repInfo);
+  }
+}
+
+uint16_t ClientRequestMsg::getExpectedSignatureLength() const {
+  std::stringstream msg;
+  auto sigManager = SigManager::getInstance();
+  uint16_t expectedSigLen{0};
+
+  if (sigManager->isClientTransactionSigningEnabled()) {
+    expectedSigLen = sigManager->getSigLength(senderId());
+    if (expectedSigLen == 0) {
+      std::stringstream msg;
+      msg << "Failed to get signature length for senderid " << senderId();
+      LOG_ERROR(GL, msg.str());
+      throw ClientSignatureVerificationFailedException(msg.str());
+    }
+  }
+  return expectedSigLen;
+}
+
+void ClientRequestMsg::validateRequest(const ReplicasInfo& repInfo, uint16_t expectedSigLen) const {
+  PrincipalId senderId = this->senderId();
+  ConcordAssert(senderId != repInfo.myId());
+  auto header = msgBody();
```

Comment:
@yuliasherman , What is the difference performance-wise? + why not just use const?
I can define:
`const auto header`
or define:
`const ClientRequestMsgHeader* header`
and define as you suggest. in all cases, a pointe is returned.

Reply:
an update. "auto &header" wont compile:
non-const lvalue reference to type
      'bftEngine::ClientRequestMsgHeader *' cannot bind to a temporary of type 'bftEngine::ClientRequestMsgHeader *'
    auto& header = msgBody();

changed to o
``const auto* header = msgBody();

Reply:
In case you use a reference, no new pointer is created. Const is for the creation of a const pointer. Not critical. I believe it has a very small difference in the performance.